src/changelogs/parse_changelogs.py

```diff
@@ -93,73 +235,54 @@ def _get_resources(self):
-                if hero not in hero_changelogs:
-                    hero_changelogs[hero] = {}
-                hero_changelogs[hero][date] = changes
-
-        for hero_name, changelog in hero_changelogs.items():
-            json_utils.write(
-                self.OUTPUT_CHANGELOGS + f'/hero/{hero_name}.json',
-                self._sort_object_by_date_key(changelog),
-            )
-
-        item_changelogs = {}
-        for date, changelog in self.changelogs_by_date.items():
-            for item, changes in changelog['Items'].items():
-                if item not in item_changelogs:
-                    item_changelogs[item] = {}
-                item_changelogs[item][date] = changes
-
-        for item_name, changelog in item_changelogs.items():
-            json_utils.write(
-                self.OUTPUT_CHANGELOGS + f'/item/{item_name}.json',
-                self._sort_object_by_date_key(changelog),
-            )
-
-    # Convert changelogs to an array of rows, with the plan to upload
-    # them to a database (TODO)
-    def _create_changelog_db_data(self):
-        rows = []
-        for date, changelog in self.changelogs_by_date.items():
-            for header, log_groups in changelog.items():
-                for group_name, logs in log_groups.items():
-                    for index, log in enumerate(logs):
-                        changelog_row = {
-                            'id': f'{header}-{group_name}-{index}',
-                            'changelog': log,
-                            'resource_key': group_name,
-                            'resource_type': header,
-                            'patch_version': '',
-                            'timestamp': date,
-                        }
-                        rows.append(changelog_row)
-
-        return rows
-
-    def _sort_object_by_date_key(self, changelogs):
-        sorted_keys = changelogs.keys()
-        sorted_changelogs = {}
-        for key in reversed(sorted_keys):
-            sorted_changelogs[key] = changelogs[key]
-
-        return sorted_changelogs
+    def is_hero(self, tag):
+        """
+        Returns True if the tag is a hero name i.e. Abrams
+        """
+        for hero_key, hero_data in self.heroes.items():
+            hero_name = hero_data['Name']
+            if tag == hero_name:
+                return True
+
+        return False
+
+    def _write_tag_tree(self, tag_tree):
+        # Add <hero>, <ability>, <item> etc. to tag tree
+        # to show where instance tags would appear (such as Abrams, Basic Magazine, Siphon Life)
+        tag_tree['Hero']['<Hero Name>'] = {}
+        tag_tree['Hero']['<HeroLab Hero Name>'] = {'HeroLab <HeroLab Hero Name>': {}}
+        tag_tree['Hero']['Ability']['<Ability Name>'] = {}
+        tag_tree['Item']['Weapon Item']['<Weapon Item Name>'] = {}
+        tag_tree['Item']['Vitality Item']['<Vitality Item Name>'] = {}
+        tag_tree['Item']['Spirit Item']['<Spirit Item Name>'] = {}
```

Comment:
Is anything stopping us from just including all of this in `tags.py` directly?

Reply:
Hadn't thought about it, you absolutely right. Moved _write_tag_tree to tags.py.